### PR TITLE
ci: deprecates CircleCI test job

### PR DESCRIPTION
**Pull request by overvenus**
###  What have you changed?

Deprecates CircleCI test job.

###  What is the type of the changes?

- Misc (other changes)

### CODE REVIEW

**Comment by breeswish** [comment]
Why not just remove it?

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,148 +1,8 @@
 version: 2
```
[the remark this answers is not in the document]
Comment (siddontang) [reply .circleci/config.yml]: PTAL @zhouqiang-cl

[the remark this answers is not in the document]
Comment (breeswish) [reply .circleci/config.yml]: +1 to check it in our CI directly.

[the remark this answers is not in the document]
Comment (jespino) [reply .circleci/config.yml]: Looks like this never was added to the CI, I created a PR to fix the spelling errors reported by codespell here: #10170

The codespell line used for that was `codespell -L crate,nd,ser,hist,ths,ba,hel,te,ans,tThe src components cmd fuzz tests` I guess merging that PR and adding that command executing to the jenkins should be enough.

**Pull request merged**